jest/index.d.ts

```diff
@@ -83,6 +83,8 @@ declare namespace jest {
     function useFakeTimers(): typeof jest;
     /** Instructs Jest to use the real versions of the standard timer functions. */
     function useRealTimers(): typeof jest;
+    /** Creates a mock function similar to jest.fn but also tracks calls to object[methodName] */
+    function spyOn(object: any, method: string): typeof jest;
```

Comment:
I'd suggest this
```ts
function spyOn<T extends {}, M extends keyof T>(object: T, method: M): MockInstance<T[M]>;
```
Also return type ```typeof jest``` is wrong. ```jest.spyOn``` returns mock/spy instance not the global jest object

Better ```T extends object``` but this require TS 2.2 and PR won't be merged until 4-6 weeks passed